packages/clients/textLocator/src/plugins/GeometrySearch/utils/makeTreeView.ts

```diff
@@ -52,29 +52,39 @@ export const makeTreeView = (
   const byTextTreeViewItems: TreeViewItem[] = Object.entries(
     titleLocationFrequency
   )
-    .map(
-      ([title, locations]): TreeViewItem => ({
-        id: title, // TODO additional ID requested; switch when available
-        name: title,
-        count: Object.values(locations).reduce((acc, curr) => acc + curr),
-        type: 'text',
-        children: Object.entries(locations)
-          .map(([locationId, count]): TreeViewItem => {
-            const feature = featureCollection.features.find(
-              (feature) => feature.id === locationId
-            )
-            return {
-              id: locationId,
-              name: featureCollection.features.find(
+    .reduce(
+      (
+        accumulator,
+        [literatureId, { title, location_frequency: locationFrequency }]
+      ): TreeViewItem[] => [
+        ...accumulator,
+        {
+          id: literatureId,
+          name: title,
+          count: Object.entries(locationFrequency).reduce(
+            (accumulator, current) => accumulator + current[1],
+            0
+          ),
```

Comment:
The missing initializer `0` may lead to errors, should we ever have a situation where `locationFrequency` has no entries. Have you checked whether this may occur? I'd add the `0` just to be safe.

Reply:
Haven't found any situation where that is the case, but added the 0 for safety, as it doesn't really bring in any problems